LR/07/src/functional.cpp

```diff
@@ -1,40 +1,83 @@
 #include "functional.h"
+#include "tcpserver.h"
 
 /**
  * @brief Конструктор
  * @param parent - родитель этого класса, базовый QObject
+ * @param linkLab - ссылка на репозиторий GitHub
  */
-Functional::Functional(QObject *parent)
-    : QObject(parent)
+Functional::Functional(QString linkLab, QObject *parent) : QObject(parent)
 {
     manager = new QNetworkAccessManager();
+    link = linkLab;
+    linkChange();
+
+    connect(manager, &QNetworkAccessManager::finished, this, &Functional::slotCheckRepo);
+}
+
+/**
+ * @brief Метод приводит ссылку к нужному виду
+ * @return void
+ */
+void Functional::linkChange()
+{
+    link.insert(8, "api.");
+    link.insert(23, "repos/");
+    link.push_back("/contents");
 }
 
 /**
  * @brief Метод посылает GET запрос на GitHub и получает ответ в формате Json
  * @return void
  */
-void Functional::getContentFromGithub()
+void Functional::getDataFromGithub()
 {
-    QUrl url("/*link*/");
+    QUrl url(link);
     QNetworkRequest request;
     request.setUrl(url);
     reply = manager->get(request);
 }
 
 /**
- * @brief Метод получает ссылку на файл с кодом,
- *        вызывает метод GetContentFromGitHub(),
- *        получает листинг кода в формате Json
+ * @brief Слот обрабатывает Json с GitHub'a
  * @return void
  */
-void Functional::getLinkToFile()
+void Functional::slotCheckRepo()
 {
-    QJsonDocument catalog;
-    QJsonObject currentPartCatalog;
-    QString t = reply->readAll();
-    qDebug() << t;
-    // Незакончен
+    QString tempString;
+    QByteArray jsonData, jsonData2;
+    QJsonArray array;
+    QString chek = "cpp";
+
+    if  (reply->error() == QNetworkReply::NoError) {
+        jsonData = reply->readAll();
+        jsonData2 = jsonData;
+        array = QJsonDocument::fromJson(jsonData).array();
+
+        /* Проверяем что в репозитории находится только один файл */
+        if (array.size() == 1) {
+            for (const QJsonValue& value : array) {
+                tempString = value.toObject() ["name"].toString();
+            }
+
+            /* Проверяем что это файл .cpp */
+            if (tempString.indexOf(chek, -5) != -1) {
+
+                /* Извлекаем ссылку */
+                array = QJsonDocument::fromJson(jsonData).array();
+                for (const QJsonValue& value : array) {
+                    tempString = value.toObject() ["path"].toString();
+                    link.push_back("/");
+                    link.push_back(tempString);
+
+                    /* Получаем данные файла #218 */
+                }
+            } else
+                  qDebug() << "Error::TypeFile";
+        } else
+              qDebug() << "Error::NumberOfRepo";
+    } else
+          qDebug() << reply->error();
```

Comment:
добавь сюда триггер сигнала, чтобы клиенту пришло понятное объяснение:
```suggestion
          QString message = reply->error();
          emit Gateway::systemError(message);
          qCrititcal() << message;
```